### ISSUE TITLE
malloc.h Mac OSX Build Error(s)

### ISSUE THREAD

**Issue by EthanMichalicek** [BinomialLLC/basis_universal#206]
1.13 doesn't build on Mac by default (at least not on my machine). I was able to get a successful build after changing 3 lines.

In `basisu_containers.h`:
`#include <malloc.h>` to `#include <malloc/malloc.h>` on line 8.

In `basisu_containers_impl.h`:
`malloc_usable_size` to `malloc_size` on lines 63 and 90.

It's totally possible I missed something important in the build process that gets rid of this issue, but I am not sure, since it built and ran just fine after making those changes.

**Comment by MarkCallow** [comment]
It compiled without issue for me in Xcode 12.4 on macOS Catalina 10.15.7.

Are you using the latest code?  In the current master the include you refer to is surrounded by an `ifdef __linux__` so on macOS it won't even be trying to include `malloc.h`.

Just in case it is relevant, though I doubt it, I am not using the CMakeLists in this repo. I am building the code as part of KTX-Software using its CMakeLists.txt.

**Comment by EthanMichalicek** [comment]
Hmm, I downloaded the 1.13 release source, so that might be out of date. Yes, I did build with the CMakeLists included here. Seems like it's specifically an issue with me not having the correct `malloc.h` on my machine (from my limited research, it seems like there's a _slightly_ different version for Mac OSX, but I'm not super familiar with this file/C++). But it seems like if I had built from `master` it would have solved my problem.
